### DISCUSSION TITLE
Want to see your feature requests on our 2025 roadmap?

### DISCUSSION THREAD

**Discussion by joebudi** [Budibase/budibase#15211] [Polls]
We're using this poll to help determine what features we should build in Q1/Q2 2025 and potentially source users for further research.

**Comment by feurer98** [comment]
ideally everything, of course :)

**Reply by joebudi** [reply]
From highest to lowest, please prioritize the features above 🔢

**Reply by feurer98** [reply]
1. Offline
2. automation debuging
3. state variables
4. i18n
5. python
6. domain
7. AI

**Comment by Timmiej93** [comment]
In all honesty: Better documentation. There is hardly any information on using the API for custom component development for example. I'm sure the info there is plenty for those who are already deeply familiar with BB, but this is something that would steer a developer who's new to BB away from this software.

**Reply by mikesealey** [reply]
@Timmiej93 is there something in particular you're struggling with?

**Reply by Timmiej93** [reply]
Basically the entire internal API. I've resorted to using `API.get({ url: 'api/...' })` for basically everything, but I can't imagine that's how it's supposed to be used. There's a mention of `/server/src/api/controllers` and `server/src/api/routes` in the custom component documentation, but that barely helped me (also, both links are broken, they lead to the `develop` branch instead of `master`).

I get that this is no-code software, so the main target is a no-code audience, but that no-code audience does need (/want) features that have to be created by programmers.

**Comment by benaunt** [comment]
1. View state variables/queries - An easy way to view available state variables and queries within design
2. PWA / Offline apps 

For the variables can you add the ability to add a comment against them, then some mechanism to export it all for system documentation, would be great if that comment function could put but against any component in app and exported with screen name, component, comment :).

Some additional features

- More customisation options for the header/navigation in an app
- Ability to customise font colours globally in an app
- Ability to apply CSS/JS globally across all screens in an app
- Improved dev ops processes, export app changes to git), ability to lifecycle an app between different running Budibase instances with an expanded environment variable function to adjust the values based on instance
- Ability to specify what related tables data is pulled from when using relationship pickers, currently if a related table has additional relationships the app performance will grind to a halt. As each option in the picker will pull the ID's of every record for every relationship of the primary table the picker pulls data from. Often none of the related data is required all the time especially for forms.
- Expanded actions in the automation module, maths, state machines, user interaction tied to "forms" etc.

**Reply by joebudi** [reply]
Thank you for your suggestions. They're all valid. We're currently planning our first cycle and we'll inform everyone what we're working on next week.
